### PR TITLE
add logic to delete realm role

### LINKED ISSUE #135: add logic to delete realm role

### PR DESCRIPTION
**Pull request by jenny-s51**
towards #94 . closes #135 and #131

### CODE REVIEW

**Comment by jenny-s51** [comment]
at the moment, the realm role can be successfully deleted from the server, but the table is only updated on a manual reload... @edewit would you mind clarifying how you were able to get the table component to update in `ClientList` when a client is deleted? I've been looking into how to force a reload for a component but wanted to ask you if there is a better way because it looks like you didn't end up implementing a forced update at all

**Comment by edewit** [comment]
@jenny-s51 Right, because also for the client list this doesn't work yet.

**Comment by ssilvert** [comment]
> @jenny-s51 Right, because also for the client list this doesn't work yet, but I introduced it in #143

@jenny-s51 #143 is now merged.  Please update for that and also fix typescript errors.

**Comment by jenny-s51** [comment]
@ssilvert @edewit this PR is all set and ready for review. Do you have any suggestions as to how to go about fixing the check-types error? The test passes locally for me, and alerts in `RoleList` are implemented exactly how @edewit implemented them in `ClientsList.tsx`. 

```
Error: src/realm-roles/RoleList.tsx(35,9): error TS2488: Type 'AlertProps' must have a '[Symbol.iterator]()' method that returns an iterator.
```

File: src/realm-roles/RealmRolesSection.tsx [hunk 1]
```diff
@@ -1,46 +1,70 @@
...
-        >
-          <DataLoader loader={loader}>
-            {(roles) => <RolesList roles={roles.data} />}
-          </DataLoader>
-        </TableToolbar>
+        {!roles && (
+          <div className="pf-u-text-align-center">
+            <Spinner />
+          </div>
+        )}
```
Comment (srambach) [inline src/realm-roles/RealmRolesSection.tsx:44]: I think this should be a variant="light" page section. And I think the spinner would work nicely in a <Bullseye> layout in place of this div.

File: src/realm-roles/add/NewRoleForm.tsx [hunk 2]
```diff
@@ -75,6 +77,7 @@ export const NewRoleForm = () => {
               render={({ onChange, value }) => (
                 <TextArea
                   type="text"
+                  validated={value.length >= 255 ? "error" : "default"}
```
Comment (edewit) [inline src/realm-roles/add/NewRoleForm.tsx:80]: you could use the `errors` object from the "react-hook-form" instead right? No need to check the length again.

File: src/realm-roles/RealmRolesSection.tsx [hunk 3]
```diff
@@ -1,46 +1,70 @@
...
 import { HttpClientContext } from "../context/http-service/HttpClientContext";
 import { RoleRepresentation } from "../model/role-model";
 import { RolesList } from "./RoleList";
 import { RealmContext } from "../context/realm-context/RealmContext";
 import { ViewHeader } from "../components/view-header/ViewHeader";
+import { PaginatingTableToolbar } from "../components/table-toolbar/PaginatingTableToolbar";
 
 export const RealmRolesSection = () => {
+  const [max, setMax] = useState(10);
+  const [first, setFirst] = useState(0);
```
Comment (edewit) [inline src/realm-roles/RealmRolesSection.tsx:15]: does the roles endpoint support pagination?

**Comment by edewit** [comment]
what is the difference between this one and #158 ?

**Comment by ssilvert** [comment]
Closing this.  @jenny-s51 will submit a new PR combining it with #158.

File: src/realm-roles/RealmRolesSection.tsx (hunk 3 above)
Comment (jenny-s51) [reply src/realm-roles/RealmRolesSection.tsx:15]: could you kindly clarify what you mean by this @edewit ?

File: src/realm-roles/add/NewRoleForm.tsx (hunk 2 above)
Comment (jenny-s51) [reply src/realm-roles/add/NewRoleForm.tsx:80]: nice catch, thank you

File: src/realm-roles/RealmRolesSection.tsx (hunk 3 above)
Comment (edewit) [reply src/realm-roles/RealmRolesSection.tsx:15]: I does support pagination, sorry about that

**Pull request closed without merging**